pyVmomi/VmomiSupport.py

```diff
@@ -460,6 +460,8 @@ def __getattr__(self, name):
    def __eq__(self, other):
       if other is None:
          return False
+      elif type(other) is str:
```

Comment:
My understanding is that the solution won't work when comparing to unicode in python2 or bytes in python3 as they are not str.